addons/api/mirage/factories/host-catalog.js

```diff
@@ -16,6 +17,26 @@ export default factory.extend({
       'host-sets': ['create', 'list'],
     };
   },
+  attributes() {
+    if (this.plugin) {
+      switch (this.plugin.name) {
```

Comment:
Would this work as a way to eliminate the outer conditional?
```suggestion
      switch (this.plugin?.name) {
```